src/main/java/com/airbnb/domain/booking/service/BookingService.java

```
@@ -37,17 +35,37 @@ public class BookingService {
     private final BookingRepository bookingRepository;
     private final MemberRepository memberRepository;
     private final AccommodationRepository accommodationRepository;
-    private final AmountCalculationUtil amountCalculationUtil;
-    private final PolicyService policyService;
+    private final FeePolicyRepository feePolicyRepository;
+    private final DiscountPolicyRepository discountPolicyRepository;
 
     @Transactional
-    public BookingResponse create(String guestKey, Long accommodationId, BookingCreateRequest request) {
-        Member guest = memberRepository.findByEmail(guestKey).orElseThrow();
+    public BookingResponse create(Long guestId, Long accommodationId, BookingCreateRequest request) {
+        if (!request.isCheckInAfterToday()) {
+            throw new IllegalArgumentException("오늘 이후의 날짜만 예약이 가능합니다.");
+        }
+
+        if(!request.isCheckOutAfterCheckIn()) {
+            throw new IllegalArgumentException("체크아웃은 체크인 이후여야 합니다.");
+        }
```

역시 .... 이럴 줄 알았지만... 반영 못 했다고 하셨으니까 🤣 

fail fast 도 좋은 방법 중 하나라고 생각해요
서비스 오기전, 컨트롤러에서 해도 되죵

아마 검증 로직을 한 군데서 관리 하고 싶단 생각이 었을 거 예요.
그런 경우에는 .... 😉  전 미팅 때 말했었어요 ㅋㅋㅋ

---

사실 지금 기억이 약간... 흐릿한데요 😅

찾아보니 커스텀 validator를 생성해서 annotation과 함께 사용하거나... 하는 방식도 있겠지만
aop 사용하는 방식 좋아보여요
시도해 보겠습니다...!